File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/database/DbHelper.java
package ru.avystavkin.githubgists.database;

import java.util.List;

import ru.avystavkin.githubgists.models.local.Gist;
import ru.avystavkin.githubgists.models.local.User;
import ru.avystavkin.githubgists.repository.GistDao;

public class DbHelper {
    private GistDao mGistDao;
    public DbHelper(AppDatabase appDatabase) {
        mGistDao = appDatabase.gistDao();
    }

    public void insert(List<Gist> gists) {
        mGistDao.clearUserTable();
        mGistDao.clearGistTable();

        for (Gist gist : gists) {
            mGistDao.insert(gist.getUser());
            mGistDao.insert(gist);
        }
    }

    public List<Gist> getGists() {
       return mGistDao.getGists();
    }

    public List<User> getPopularUsers(int count) {
        return mGistDao.getPopularUsers(count);
    }
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/database/DbHelper.kt
package ru.avystavkin.githubgists.database

import ru.avystavkin.githubgists.models.local.Gist
import ru.avystavkin.githubgists.models.local.User
import ru.avystavkin.githubgists.repository.GistDao

class DbHelper(appDatabase: AppDatabase) {
    private val mGistDao: GistDao = appDatabase.gistDao()

    val gists: List<Gist> = mGistDao.gists

    fun insert(gists: List<Gist>) {
        mGistDao.clearUserTable()
        mGistDao.clearGistTable()

        for (gist in gists) {
            mGistDao.insert(gist.user)
            mGistDao.insert(gist)
        }
    }

    fun getPopularUsers(count: Int): List<User> {
        return mGistDao.getPopularUsers(count)
    }
}
